pandas/io/json/_normalize.py

```diff
@@ -280,8 +280,6 @@ def _pull_field(js, spec):
     lengths = []
 
     meta_vals = defaultdict(list)  # type: DefaultDict
-    if not isinstance(sep, str):
```

Comment:
I suppose this would break code if someone tried to provide a non-str sep but that's not documented and unclear why someone would try that anyway

Reply:
could it be a remnant of py2 compatibility? for public methods should we allow bytes in all places where a string is expected?

Reply:
Not sure but on closer inspection doesn't seem to work in general case anyway:

```python
>>> json_normalize({ 'foo' : { 'bar' : 0 } }, sep=b'.')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/williamayd/clones/pandas/pandas/io/json/_normalize.py", line 266, in json_normalize
    data = nested_to_record(data, sep=sep, max_level=max_level)
  File "/Users/williamayd/clones/pandas/pandas/io/json/_normalize.py", line 106, in nested_to_record
    new_d.update(nested_to_record(v, newkey, sep, level + 1, max_level))
  File "/Users/williamayd/clones/pandas/pandas/io/json/_normalize.py", line 91, in nested_to_record
    newkey = prefix + sep + k
TypeError: can only concatenate str (not "bytes") to str
```

So just seems like a mistake